File: car_rental_angular/src/app/modules/admin/service/admin.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { StorageService } from '../../../auth/services/storage/storage.service';
import { Observable } from 'rxjs';

const BASE_URL = ["http://localhost:8080"];

@Injectable({
  providedIn: 'root'
})
export class AdminService {

  constructor( private http: HttpClient) { }

  postCar(carDto:any): Observable<any> {
    return this.http.post(BASE_URL+"/api/admin/car",carDto,{
      headers:this.createAuthorizationHeader()
    });
  }

  getAllCars(): Observable<any> {
    return this.http.get(BASE_URL+"/api/admin/cars",{
      headers:this.createAuthorizationHeader()
    });
  }

  deleteCar(id:number):Observable<any>{
    return this.http.delete(BASE_URL+ "/api/admin/car/" + id,{
      headers:this.createAuthorizationHeader()
    });
  }

  getCarById(id:number):Observable<any>{
    return this.http.get(BASE_URL+ "/api/admin/car/" + id,{
      headers:this.createAuthorizationHeader()
    });
  }

    updateCar(carId:number,carDto:any):Observable<any>{
    return this.http.put(BASE_URL+"/api/admin/car/"+carId,carDto,{
      headers:this.createAuthorizationHeader()
    });
  }

  getCarBookings(): Observable<any> {
    return this.http.get(BASE_URL+"/api/admin/car/bookings",{
      headers:this.createAuthorizationHeader()
    });
  }

   changeBookingStatus(bookingId:number,status :string): Observable<any> {
    return this.http.get(BASE_URL+`/api/admin/car/booking/${bookingId}/`+status ,{
     headers:this.createAuthorizationHeader()
    });
  }

   searchCar(searchCarDto:any): Observable<any> {
    return this.http.post(BASE_URL+"/api/admin/car/search",searchCarDto,{
      headers:this.createAuthorizationHeader()
    });
  }



  createAuthorizationHeader():HttpHeaders{
    let authHeaders: HttpHeaders = new HttpHeaders();
    return authHeaders.set(
      'Authorization',
      'Bearer ' + StorageService.getToken()
    );
    }


}

File: CarRentingSystem/src/main/java/com/example/carrentingsystemSpringBootAppliction/services/admin/AdminService.java
package com.example.carrentingsystemSpringBootAppliction.services.admin;

import com.example.carrentingsystemSpringBootAppliction.dto.BookACarDto;
import com.example.carrentingsystemSpringBootAppliction.dto.CarDto;
import com.example.carrentingsystemSpringBootAppliction.dto.CarDtoListDto;
import com.example.carrentingsystemSpringBootAppliction.dto.SearchCarDto;

import java.io.IOException;
import java.util.List;


public interface AdminService {

    boolean postCar(CarDto carDto) throws IOException;

    List<CarDto> getAllCars();

    void deleteCar(Long id);

    CarDto getCarById(Long id);

    boolean updateCar(Long id,CarDto carDto) throws IOException;

    List<BookACarDto> getbookings();

    boolean changeBookingStatus(Long bookingId, String status);

    CarDtoListDto searchCars(SearchCarDto searchCarDto);
}
